### PR TITLE
ONNX documentation

### PR DESCRIPTION
**Pull request by mfuntowicz**
- Rename the current **torchscript.rst** to **serialization.rst**
- Move torchscript documentation into a subsection of the above **serialization.rst**
- Introduce documentation for ONNX/ONNXRuntime in the above section

Signed-off-by: Morgan Funtowicz <funtowiczmo@gmail.com>

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Seems like some files are missing?

Are we expecting more than 2 model_export subitems? If no I'd leave that in the section before, but if we think there will be a lot more, I don't mind a dding a section.

**Review by LysandreJik — APPROVED** [review]
I like it, very nice!

**Review by sgugger — APPROVED** [review]
Yes, it's good to go! Thanks!

(nit for doc writing: remember to put two underscores after your links as it can cause some problems if two links have the same labels. Here it's harmless as the labels are long.)

File: docs/source/serialization.rst [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+
+ONNX / ONNXRuntime
+==============================================
+
+Projects ONNX (Open Neural Network eXchange) and ONNXRuntime (ORT) are part of an effort from leading industries in the AI field
+to provide a unified and community-driven format to store and, by extension, efficiently execute neural network leveraging a variety
+of hardware and dedicated optimizations.
+
+Starting from transformers v2.10.0 we partnered with ONNX Runtime to provide an easy export of transformers models to
+the ONNX format. You can have a look at the effort by looking at our join blog post `Accelerate your NLP pipelines using
```
Comment (LysandreJik) [inline docs/source/serialization.rst:13]:
```suggestion
the ONNX format. You can have a look at the effort by looking at our joint blog post `Accelerate your NLP pipelines using
```

File: docs/source/serialization.rst [hunk 2]
```diff
@@ -1,5 +1,44 @@
...
+Hugging Face Transformers and ONNX Runtime <https://medium.com/microsoftazure/accelerate-your-nlp-pipelines-using-hugging-face-transformers-and-onnx-runtime-2443578f4333>`_.
+
+Exporting a model is done through the script `convert_graph_to_onnx.py` at the root of the transformers sources.
+The following command shows how easy it is to export a BERT model from the library, simply run:
+
+.. code-block:: bash
+
+    python convert_graph_to_onnx.py --framework <pt, tf> --model bert-base-cased bert-base-cased.onnx
+
+The conversion tool works for both PyTorch and Tensorflow models and ensure:
```
Comment (LysandreJik) [inline docs/source/serialization.rst:23]:
```suggestion
The conversion tool works for both PyTorch and Tensorflow models and ensures:
```

File: docs/source/serialization.rst [hunk 3]
```diff
@@ -1,5 +1,44 @@
...
+    * The model and its weight are correctly initialized from the Hugging Face's model hub or a local checkpoint.
```
Comment (LysandreJik) [inline docs/source/serialization.rst:24]:
```suggestion
    * The model and its weights are correctly initialized from the Hugging Face's model hub or a local checkpoint.
```

Comment (LysandreJik) [inline docs/source/serialization.rst:24]: either
```suggestion
    * The model and its weight are correctly initialized from Hugging Face's model hub or a local checkpoint.
```
or
```suggestion
    * The model and its weight are correctly initialized from the Hugging Face model hub or a local checkpoint.
```

File: docs/source/serialization.rst [hunk 4]
```diff
@@ -1,5 +1,44 @@
...
+.. note::
+    Currently, inputs and outputs are always exported with dynamic sequence axes preventing some optimizations
+    on the ONNX Runtime. If you would like to see such support for fixed-length inputs/outputs, please
+    open up an issue on transformers.
+
+
+Also, the conversion tool supports different options which let you tune the behavior of the generated model:
+    * Change the target opset version of the generated model: More recent opset generally supports more operator and enables faster inference.
+    * Export pipeline specific prediction heads: Allow to export model along with its task-specific prediction head(s).
+    * Use the external data format (PyTorch only): Let you export model which size is above 2Gb (`More info <https://github.com/pytorch/pytorch/pull/33062>`_).
```
Comment (LysandreJik) [inline docs/source/serialization.rst:37]:
```suggestion
    * Use the external data format (PyTorch only): Lets you export model which size is above 2Gb (`More info <https://github.com/pytorch/pytorch/pull/33062>`_).
```

**Review by LysandreJik — APPROVED** [review]
Final checks :)

**Comment by mfuntowicz** [comment]
Merging as the failures are not related to the changeset introduced in this PR

**Pull request merged**